packages/apollo-server-fastify/src/ApolloServer.ts

```diff
@@ -0,0 +1,145 @@
+import { renderPlaygroundPage } from '@apollographql/graphql-playground-html';
+import { Accepts } from 'accepts';
+import {
+  ApolloServerBase,
+  PlaygroundRenderPageOptions,
+} from 'apollo-server-core';
+import { FastifyInstance, FastifyReply, FastifyRequest } from 'fastify';
+import { IncomingMessage, OutgoingMessage } from 'http';
+import { processRequest as processFileUploads } from '@apollographql/apollo-upload-server';
+import { graphqlFastify } from './fastifyApollo';
+
+const fastJson = require('fast-json-stringify');
+
+export interface ServerRegistration {
+  app: FastifyInstance;
+  path?: string;
+  cors?: object | boolean;
+  onHealthCheck?: (req: FastifyRequest<IncomingMessage>) => Promise<any>;
+  disableHealthCheck?: boolean;
+}
+
+const stringifyHealthCheck = fastJson({
+  type: 'object',
+  properties: {
+    status: {
+      type: 'string',
+    },
+  },
+});
+
+export class ApolloServer extends ApolloServerBase {
+  protected supportsSubscriptions(): boolean {
+    return true;
+  }
+
+  protected supportsUploads(): boolean {
+    return true;
+  }
+
+  public async applyMiddleware({
+    app,
+    path,
+    cors,
+    disableHealthCheck,
+    onHealthCheck,
+  }: ServerRegistration) {
+    await this.willStart();
+
+    if (!path) path = '/graphql';
+
+    this.graphqlPath = path;
+
+    app.register(require('fastify-accepts'));
```

Comment:
All the following register and route definition are better wrapped in a standalone plugin.